ee/localserver/krypto-ec-middleware.go

```diff
@@ -153,20 +176,44 @@ func (e *kryptoEcMiddleware) Wrap(next http.Handler) http.Handler {
 		var cmdReq v2CmdRequestType
 		if err := json.Unmarshal(challengeBox.RequestData(), &cmdReq); err != nil {
 			traces.SetError(span, err)
-			level.Debug(e.logger).Log("msg", "unable to unmarshal cmd request", "err", err)
+			e.slogger.Log(r.Context(), slog.LevelError,
+				"unable to unmarshal cmd request",
+				"err", err,
+			)
+
 			w.WriteHeader(http.StatusUnauthorized)
 			return
 		}
 
+		// set the kolide session id if it exists, this also the saml session id
+		kolideSessionId, ok := cmdReq.CallbackHeaders[multislogger.KolideSessionIdKey.String()]
+		if ok && len(kolideSessionId) > 0 {
+			span.SetAttributes(attribute.String(multislogger.KolideSessionIdKey.String(), kolideSessionId[0]))
+			r = r.WithContext(context.WithValue(r.Context(), multislogger.KolideSessionIdKey, kolideSessionId[0]))
```

Comment:
I think it would be less awkward if the expected header name was a constant here, it should not overload `multislogger.KolideSessionIdKey`. That brings in unneeded coupling

Reply:
I've updated it so that we define the header key here as a constant. However, we still use the contextKey iota in multislogger to set context value

Reply:
Yah. That part feels awkward, but I don't know enough to know how to fix it. We are, ultimately, trying to set internal state in context that multislogger uses. So it has to come from there.

But we should not stash http headers there